### PR TITLE
Pass context 2

### PR DESCRIPTION
**Pull request by reshnm**
**How to categorize this PR?**

This PR is a follow-up of PR #372 since the original PR can't no longer be modified. 

/area deployers
/priority 3

**What this PR does / why we need it**:

The context reference specified in the installation is now passed to the subinstallations, execution and deploy items.
With that it is now possible for the deployers to use namespaced secrets to access oci or other artifacts.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature operator
The context reference is not properly passed to all DeployItems and subinstallations.
With that the deployers are now able to use the secrets defined in the Context to access remote artifacts.
This is especially useful for multi-tenant scenarios where everything should be isolated by the namespace.

```

### CODE REVIEW

**Review by Diaphteiros — CHANGES_REQUESTED** [review]
I made a few comments. Most of them are just minor nits referring to wrong comments on the new context fields (comment still talks about installations, probably a copy-paste error). Some of the places where I marked this (and those where I didn't mark this) are auto-generated during `make generate`, I think.

File: pkg/deployer/manifest/controller.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ type deployer struct {
 	hooks      extension.ReconcileExtensionHooks
 }
 
-func (d *deployer) Reconcile(ctx context.Context, di *lsv1alpha1.DeployItem, target *lsv1alpha1.Target) error {
+func (d *deployer) Reconcile(ctx context.Context, lsCtx *lsv1alpha1.Context, di *lsv1alpha1.DeployItem, target *lsv1alpha1.Target) error {
 	logger := d.log.WithValues("resource", types.NamespacedName{Name: di.Name, Namespace: di.Namespace}.String())
 	manifest, err := New(logger, d.lsClient, d.hostClient, &d.config, di, target)
```
[the remark this answers is not in the document]
Comment (reshnm) [reply pkg/deployer/manifest/controller.go]: It seems that it is the latter. The context is not used for handling kubernetes manifests.
I marked the parameter as unused.

File: pkg/landscaper/crdmanager/crdresources/landscaper.gardener.cloud_deployitems.yaml [hunk 2]
```diff
@@ -40,6 +40,9 @@ spec:
                 type: object
                 x-kubernetes-embedded-resource: true
                 x-kubernetes-preserve-unknown-fields: true
+              context:
+                description: Context defines the current context of the installation.
```
Comment (Diaphteiros) [inline pkg/landscaper/crdmanager/crdresources/landscaper.gardener.cloud_deployitems.yaml:44]:
```suggestion
                description: Context defines the current context of the deployitem.
```

File: pkg/landscaper/crdmanager/crdresources/landscaper.gardener.cloud_executions.yaml [hunk 3]
```diff
@@ -32,6 +32,9 @@ spec:
           spec:
             description: Spec defines a execution and its items
             properties:
+              context:
+                description: Context defines the current context of the installation.
```
Comment (Diaphteiros) [inline pkg/landscaper/crdmanager/crdresources/landscaper.gardener.cloud_executions.yaml:36]:
```suggestion
                description: Context defines the current context of the execution.
```

**Review by Diaphteiros — CHANGES_REQUESTED** [review]
Found another two, seems they didn't get overwritten by `make generate`...

**Review by Diaphteiros — APPROVED** [review]
/lgtm

**Pull request merged**